actions/views/channel_sidebar.ts

```diff
@@ -58,33 +58,37 @@ export function createCategory(teamId: string, displayName: string, channelIds?:
 
 // moveChannelInSidebar moves a channel to a given category in the sidebar, but it accounts for when the target index
 // may have changed due to archived channels not being shown in the sidebar.
-export function moveChannelInSidebar(categoryId: string, channelId: string, targetIndex: number) {
-    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
-        const newIndex = adjustTargetIndexForMove(getState(), categoryId, channelId, targetIndex);
+export function moveChannelsInSidebar(categoryId: string, channelIds: string[], targetIndex: number, draggableChannelId: string) {
+    return (dispatch: DispatchFunc, getState: () => GlobalState) => {
+        const newIndex = adjustTargetIndexForMove(getState(), categoryId, channelIds, targetIndex, draggableChannelId);
 
-        return dispatch(moveChannelToCategory(categoryId, channelId, newIndex));
+        return dispatch(moveChannelsToCategory(categoryId, channelIds, newIndex));
     };
 }
 
-export function adjustTargetIndexForMove(state: GlobalState, categoryId: string, channelId: string, targetIndex: number) {
+export function adjustTargetIndexForMove(state: GlobalState, categoryId: string, channelIds: string[], targetIndex: number, draggableChannelId: string) {
     if (targetIndex === 0) {
         // The channel is being placed first, so there's nothing above that could affect the index
         return 0;
     }
 
     const category = getCategory(state, categoryId);
     const filteredChannels = makeGetChannelsForCategory()(state, category);
+    const filteredChannelIds = filteredChannels.map((channel) => channel.id);
+
+    // When dragging multiple channels, we don't actually remove all of them from the list as react-beautiful-dnd doesn't support that
+    // Account for channels removed above the insert point, except the one currently being dragged which is already accounted for by react-beautiful-dnd
+    const removedChannelsAboveInsert = filteredChannelIds.filter((channel, index) => channel !== draggableChannelId && channelIds.indexOf(channel) !== -1 && index <= targetIndex);
 
     if (category.channel_ids.length === filteredChannels.length) {
         // There are no archived channels in the category, so the targetIndex from react-beautiful-dnd will be correct
-        return targetIndex;
+        return targetIndex - removedChannelsAboveInsert.length;
```

Comment:
0/5 but perhaps we can give this a new name since it's repeated quite a few times. Maybe we rename `targetIndex` and just this `targetIndex` going forward to reduce the diff.

Reply:
What would you suggest?